jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt

```diff
@@ -398,6 +402,27 @@ class SenderVideoConstraintsMessage(val videoConstraints: VideoConstraints) : Br
     }
 }
 
+/**
+ * A message sent from the bridge to a client (sender), indicating constraints for the sender's video streams.
+ */
+class SenderVideoConstraintsMessageV2(
+    val sourceName: String,
+    val idealHeight: Int
+) : BridgeChannelMessage(TYPE) {
+
+    /**
+     * Serialize manually because it's faster than Jackson.
+     *
+     * We use the "idealHeight" format that the jitsi-meet client expects.
+     */
+    override fun createJson(): String =
+        """{"colibriClass":"$TYPE", "sourceName":"$sourceName", "idealHeight":$idealHeight}"""
+
+    companion object {
+        const val TYPE = "SenderVideoConstraintsV2"
```

Comment:
I wonder if we can choose a better name. SourceVideoConstraints? SenderSourceConstraints? SourceConstraints?

Since this needs client changes anyway, we should rename `idealHeight` as well. Maybe `maxHeight`?

Reply:
ok, I'll go with `SenderSourceConstraints` and `maxHeight`